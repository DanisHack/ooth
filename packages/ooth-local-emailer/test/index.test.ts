import emailer from '../src';

describe('ooth-local-emailer', () => {
  let sendMail = () => null;
  let emailerConfigs;
  let emailerCallbacks;

  beforeEach(() => {
    sendMail = jest.fn();
    emailerConfigs = {
      from: 'noreply@example.com',
      siteName: 'Example Site',
      url: 'http://www.example.com',
      sendMail,
    };
    emailerCallbacks = emailer(emailerConfigs);
  });

  it('sends mails on register', async () => {
    await emailerCallbacks.onRegister({
      email: 'user@gmail.com',
      verificationToken: 'xxx',
      _id: '1',
    });
    expect(sendMail.mock.calls).toMatchSnapshot();
  });

  it('sends mail on verification token', async () => {
    await emailerCallbacks.onGenerateVerificationToken({
      email: 'user@gmail.com',
      verificationToken: 'xxx',
      _id: '1',
    });
    expect(sendMail.calls).toMatchSnapshot();
  });

  it('sends mail on set email', async () => {
    await emailerCallbacks.onSetEmail({
      email: 'user@gmail.com',
      verificationToken: 'xxx',
      _id: '1',
    });
    expect(sendMail.mock.calls).toMatchSnapshot();
  });

  it('sends mail on verify', async () => {
    await emailerCallbacks.onVerify({
      email: 'user@gmail.com',
    });
    expect(sendMail.mock.calls).toMatchSnapshot();
  });

  it('sends mail on forgot password', async () => {
    await emailerCallbacks.onForgotPassword({
      email: 'user@gmail.com',
      passwordResetToken: 'xxx',
      _id: '1',
    });
    expect(sendMail.mock.calls).toMatchSnapshot();
  });

  it('sends mail on reset password', async () => {
    await emailerCallbacks.onResetPassword({
      email: 'user@gmail.com',
    });
    expect(sendMail.mock.calls).toMatchSnapshot();
  });

  it('sends mail on change password', async () => {
    await emailerCallbacks.onChangePassword({
      email: 'user@gmail.com',
    });
    expect(sendMail.mock.calls).toMatchSnapshot();
  });
});
