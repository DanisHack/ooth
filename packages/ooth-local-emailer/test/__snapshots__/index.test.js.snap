// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ooth-local-emailer sends mail on change password 1`] = `
Array [
  Array [
    Object {
      "body": "Your password for Example Site has been changed.",
      "from": "noreply@example.com",
      "html": "Your password for Example Site has been changed.",
      "subject": "Example Site password changed",
      "to": "user@gmail.com",
    },
  ],
]
`;

exports[`ooth-local-emailer sends mail on forgot password 1`] = `
Array [
  Array [
    Object {
      "body": "Reset your password for Example Site on the following page: /reset-password?token=xxx&userId=1",
      "from": "noreply@example.com",
      "html": "Reset your password for Example Site on the following page: /reset-password?token=xxx&userId=1",
      "subject": "Reset Example Site password",
      "to": "user@gmail.com",
    },
  ],
]
`;

exports[`ooth-local-emailer sends mail on reset password 1`] = `
Array [
  Array [
    Object {
      "body": "Your password for Example Site has been reset.",
      "from": "noreply@example.com",
      "html": "Your password for Example Site has been reset.",
      "subject": "Example Site password reset",
      "to": "user@gmail.com",
    },
  ],
]
`;

exports[`ooth-local-emailer sends mail on set email 1`] = `
Array [
  Array [
    Object {
      "body": "Please verify your Example Site email by opening the following url: /verify-email?token=xxx&userId=1.",
      "from": "noreply@example.com",
      "html": "Please verify your Example Site email by opening the following url: /verify-email?token=xxx&userId=1.",
      "subject": "Verify your Example Site address",
      "to": "user@gmail.com",
    },
  ],
]
`;

exports[`ooth-local-emailer sends mail on verification token 1`] = `undefined`;

exports[`ooth-local-emailer sends mail on verify 1`] = `
Array [
  Array [
    Object {
      "body": "Your Example Site address has been verified.",
      "from": "noreply@example.com",
      "html": "Your Example Site address has been verified.",
      "subject": "Example Site address verified",
      "to": "user@gmail.com",
    },
  ],
]
`;

exports[`ooth-local-emailer sends mails on register 1`] = `
Array [
  Array [
    Object {
      "body": "Thank you for joining Example Site!",
      "from": "noreply@example.com",
      "html": "Thank you for joining Example Site!",
      "subject": "Welcome to Example Site",
      "to": "user@gmail.com",
    },
  ],
  Array [
    Object {
      "body": "Please verify your Example Site email by opening the following url: /verify-email?token=xxx&userId=1.",
      "from": "noreply@example.com",
      "html": "Please verify your Example Site email by opening the following url: /verify-email?token=xxx&userId=1.",
      "subject": "Verify your Example Site address",
      "to": "user@gmail.com",
    },
  ],
]
`;
